tests/test_s3urlcache.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import logging
+import os
+import io
+import shutil
+import sys
+import unittest
+import threading
+import requests
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from dss import Config, BucketConfig
+from dss.config import Replica
+from tests.infra import start_verbose_logging, testmode
+from dss.storage.validator import S3UrlCache
+from dss.storage.validator import SizeLimitError
+from dss.util import networking
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+start_verbose_logging()
+
+KB = 1024
+MB = KB**2
+
+RANDOMDATA = os.urandom(MB)
+
+class HTTPInfo:
+    address = "127.0.0.1"
+    port = None
+    server = None
+    thread = None
+
+    @classmethod
+    def make_url(cls):
+        cls.url = f"http://{cls.address}:{cls.port}"
+
+
+def setUpModule():
+    HTTPInfo.port = networking.unused_tcp_port()
+    HTTPInfo.server = HTTPServer((HTTPInfo.address, HTTPInfo.port), GetTestHandler)
+    HTTPInfo.make_url()
+    HTTPInfo.thread = threading.Thread(target=HTTPInfo.server.serve_forever)
+    HTTPInfo.thread.start()
+
+
+def tearDownModule():
+    HTTPInfo.server.shutdown()
+
+@testmode.standalone
+class TestS3UrlCache(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        replica = Replica.aws
+        Config.set_config(BucketConfig.TEST_FIXTURE)
+        cls.blobstore, _, cls.test_fixture_bucket = Config.get_cloud_specific_handles(replica)
+        Config.set_config(BucketConfig.TEST)
+        cls.test_bucket = replica.bucket
+
+    def setUp(self):
+        self.urls_to_cleanup = []
```

Comment:
A set would be a more appropriate data structure for this.

Reply:
changed to set()